src/bin/hive_chain.rs

```diff
@@ -33,37 +37,69 @@ impl Decoder for BlockFileCodec {
     }
 }
 
+/// The inputs to the binary.
+#[derive(Parser, Debug)]
+pub struct Args {
+    /// The path to the chain file for the hive test.
+    #[clap(short, long)]
+    chain_path: PathBuf,
+    /// The relayer address.
+    #[clap(long)]
+    relayer_address: Felt,
+    /// The relayer private key.
+    #[clap(long)]
+    relayer_pk: Felt,
+}
+
+const STARKNET_RPC_URL: &str = "http://0.0.0.0:5050";
+const MAX_FELTS_IN_CALLDATA: &str = "30000";
+
 /// Inspired by the Import command from Reth.
 /// https://github.com/paradigmxyz/reth/blob/main/bin/reth/src/commands/import.rs
 #[tokio::main]
 async fn main() -> eyre::Result<()> {
-    let chain_path = Path::new(&std::env::var("CHAIN_PATH")?).to_path_buf();
-    let relayer_address = Felt::from_str(&std::env::var("RELAYER_ADDRESS")?)?;
+    let args = Args::parse();
+
+    // Get the provider
+    let provider = JsonRpcClient::new(HttpTransport::new(Url::from_str(STARKNET_RPC_URL)?));
+    let starknet_provider = StarknetProvider::new(provider);
+
+    // Set the env
+    std::env::set_var("RELAYER_PRIVATE_KEY", format!("0x{:x}", args.relayer_pk));
+    std::env::set_var("MAX_FELTS_IN_CALLDATA", MAX_FELTS_IN_CALLDATA);
+    std::env::set_var("STARKNET_NETWORK", STARKNET_RPC_URL);
+
+    // Set the chain id
+    let chain_id = starknet_provider.chain_id().await?;
+    let chain_id: u64 = (Felt::from(u64::MAX).to_bigint() & chain_id.to_bigint()).try_into()?;
```

Comment:
Here, shouldn't it be:

```rust
let chain_id: u64 = (Felt::from(u32::MAX).to_bigint() & chain_id.to_bigint()).try_into()?;
```

to be consistent with what's here

https://github.com/kkrt-labs/kakarot-rpc/blob/9ea6096ae7aa04156caf6b6c81bb8e308929effb/src/client/mod.rs#L51-L54?

Reply:
you're right, as discussed this is an error in the code and needs to be fixed.

Reply:
2**53 and not 2**32 though